### PR TITLE
[release-1.20] manifests: add gateways.securityContext option to custom gateway securityContext

### PR DESCRIPTION
**Pull request by nicole-lihui**
**Please provide a description of this PR:**
Manual cherry of https://github.com/istio/istio/pull/51234
Fixes https://github.com/istio/istio/issues/51362

### CODE REVIEW

File: manifests/charts/istio-control/istio-discovery/values.yaml [hunk 1]
```diff
@@ -517,3 +517,10 @@ base:
 istio_cni:
   enabled: false
   chained: true
+
+# Gateway Settings
+gateways:
+  # Define the security context for the pod.
+  # If unset, this will be automatically set to the minimum privileges required to bind to port 80 and 443.
+  # On Kubernetes 1.22+, this only requires the `net.ipv4.ip_unprivileged_port_start` sysctl.
+  securityContext: {}
```
[the remark this answers is not in the document]
Comment (nicole-lihui) [reply manifests/charts/istio-control/istio-discovery/values.yaml:526]: ye

**Review by hanxiaop — APPROVED** [review]

**Pull request merged**